core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -168,15 +185,23 @@ trait PartitionedDataSourceScan extends PrunedUnsafeFilteredScan {
 }
 
 private[sql] final case class ZipPartitionScan(basePlan: SparkPlan with CodegenSupport,
```

Comment:
CodgenSupport extends SparkPlan so you don't need to do this and just CodegenSupport will suffice